Added Readmes for all templates (#84)
eclipse-theia/generator-theia-extension: templates/backend/README.md

> # Backend communication example

Reviewer: Do you mind if we update the titles of each readme with title casing?

```suggestion
# Backend Communication Example
```

> The example extension demonstrates how communicate with backend services. It contains two backend service examples:

Reviewer: ```suggestion
The example extension demonstrates how to communicate with backend services. 
It contains the following two backend service examples:
```

> * `HelloBackendService`: Can be called by the client to create a "Hello World" string. The client provides a name as a parameter, the server returns the name prefixed with "Hello", e.g. "World" as a parameter returns "Hello World".

Reviewer: No strong opinion with this suggestion, just thought it read a little easier.

```suggestion
* `HelloBackendService`: Can be called by the client to create a "Hello World" string. The client provides a name as a parameter, and the server returns the name prefixed with "Hello" (ex: "World" as a parameter returns "Hello World").
```
